examples/regression/1-quickstart/plot_heteroscedastic_1d_data.py

```diff
@@ -0,0 +1,204 @@
+    )
+    y_true = f(X_true)
+    y_true_sigma = q95 * sigma * X_true
+    return X_train, y_train, X_true, y_true, y_true_sigma
+
+
+def plot_1d_data(
+    X_train: NDArray,
+    y_train: NDArray,
+    X_test: NDArray,
+    y_test: NDArray,
+    y_test_sigma: NDArray,
+    y_pred: NDArray,
+    y_pred_low: NDArray,
+    y_pred_up: NDArray,
+    ax: plt.Axes,
+    title: str,
+) -> None:
+    """
+    Generate a figure showing the training data and estimated
+    prediction intervals on test data.
+
+    Parameters
+    ----------
+    X_train : NDArray
+        Training data.
+    y_train : NDArray
+        Training labels.
+    X_test : NDArray
+        Test data.
+    y_test : NDArray
+        True function values on test data.
+    y_test_sigma : float
+        True standard deviation.
+    y_pred : NDArray
+        Predictions on test data.
+    y_pred_low : NDArray
+        Predicted lower bounds on test data.
+    y_pred_up : NDArray
+        Predicted upper bounds on test data.
+    ax : plt.Axes
+        Axis to plot.
+    title : str
+        Title of the figure.
+    """
+    ax.set_xlabel("x")
+    ax.set_ylabel("y")
+    ax.set_xlim([0, 1])
+    ax.set_ylim([0, 1])
+    ax.scatter(X_train, y_train, color="red", alpha=0.3, label="training")
+    ax.plot(X_test, y_test, color="gray", label="True confidence intervals")
+    ax.plot(X_test, y_test - y_test_sigma, color="gray", ls="--")
+    ax.plot(X_test, y_test + y_test_sigma, color="gray", ls="--")
+    ax.plot(X_test, y_pred, label="Prediction intervals")
+    ax.fill_between(X_test, y_pred_low, y_pred_up, alpha=0.3)
+    ax.set_title(title)
+    ax.legend()
+
+
+X_train, y_train, X_test, y_test, y_test_sigma = get_heteroscedastic_data()
+
+polyn_model = Pipeline(
+    [
+        ("poly", PolynomialFeatures(degree=4)),
+        ("linear", LinearRegression(fit_intercept=False)),
+    ]
+)
+polyn_model_quant = Pipeline(
+    [
+        ("poly", PolynomialFeatures(degree=4)),
+        ("linear", QuantileRegressor(
+            alpha=1e-9,
+            fit_intercept=False,
+        )),
+    ]
+)
+
+STRATEGIES = {
+    "jackknife": {"method": "base", "cv": -1},
+    "jackknife_plus": {"method": "plus", "cv": -1},
+    "jackknife_minmax": {"method": "minmax", "cv": -1},
+    "cv_plus": {"method": "plus", "cv": 10},
+    "jackknife_plus_ab": {"method": "plus", "cv": Subsample(n_resamplings=50)},
+    "conformalized_quantile_regression": {"method": "quantile", "cv": "split"},
+}
+fig, ((ax1, ax2, ax3), (ax4, ax5, ax6)) = plt.subplots(
+    2, 3, figsize=(3 * 6, 12)
+)
+axs = [ax1, ax2, ax3, ax4, ax5, ax6]
+for i, (strategy, params) in enumerate(STRATEGIES.items()):
+    if strategy == "conformalized_quantile_regression":
+        mapie = MapieQuantileRegressor(  # type: ignore
+            polyn_model_quant,
+            **params
+        )
+        X_train_split, X_calib, y_train_spit, y_calib = train_test_split(
+            X_train,
+            y_train,
+            test_size=0.5,
```

Comment:
I would go for a `test_size` of 0.3 or 0.2

Reply:
This can be done, note that it is quite important normally to have a high calibration set --> [https://arxiv.org/pdf/2107.07511.pdf] @Section 3.1.

Reply:
I think it really depends on your usecase and the trained model (i.e. neural network) but I would prefer to keep a value of 0.2-0.3 for now.

Reply:
No indeed, I have changed it to 0.3, just wanted to make you aware that they were some studies about this.